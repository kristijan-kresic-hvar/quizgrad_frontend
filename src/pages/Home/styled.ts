import styled, { StyledComponent } from 'styled-components';

export const StyledHeader: StyledComponent<'header', any, {}, never> = styled.header`
	width: 100%;
	display: flex;
	justify-content: space-between;
	gap-x: 2rem;

	@media (max-width: 768px) {
		flex-direction: column;
		gap-y: 1rem;
		gap-x: 0;
	}
`;

export const StyledSectionLeft: StyledComponent<'section', any, {}, never> = styled.section`
	flex: 1;
	width: 100%;
	padding-left: 11.25rem;
	margin-top: 4.125rem;

	@media (max-width: 768px) {
		margin-top: 0;
	}
	@media (max-width: 1200px) {
		padding-left: 0;
	}
`;

export const StyledHeading: StyledComponent<'h1', any, {}, never> = styled.h1`
	font-family: 'Roboto';
	font-style: normal;
	font-weight: 500;
	font-size: 2.3125rem;
	line-height: 2.75rem;
	color: #333333;
	margin-bottom: 2.3125rem;
	@media (max-width: 768px) {
		text-align: center;
		font-size: 1.8rem;
		& br {
			display: none;
		}
	}
`;

export const StyledHeroImg: StyledComponent<'img', any, {}, never> = styled.img`
	width: 100%;
	flex: 1;
	object-fit: contain;
	min-width: 350px;
	user-select: none;
	pointer-events: none;
`;

export const StyledBlockQuote: StyledComponent<'blockquote', any, {}, never> = styled.blockquote`
	font-family: var(--font-secondary);
	font-style: normal;
	font-weight: 400;
	font-size: 0.875rem;
	line-height: 1rem;
	color: #828282;
	padding: 0.3rem 0 0.3rem 0.75rem;
	border-left: 0.073rem solid #333333;

	@media (max-width: 768px) {
		text-align: center;
	}
`;

export const StyledCtaButton: StyledComponent<'button', any, {}, never> = styled.button`
	background: linear-gradient(95.34deg, #fcc822 0%, #ffcd2e 100%);
	box-shadow: 0px 10.4502px 23.2228px -6.96683px #fbe18f;
	font-family: var(--font-primary);
	font-style: normal;
	font-weight: 500;
	font-size: 0.875rem;
	line-height: 1.3125rem;
	color: #ffffff;
	border: none;
	outline: none;
	padding: 0.5rem 1rem;
	cursor: pointer;
	transition: background 0.2s ease-in-out;

	&:hover {
		background: #ffffff;
		color: var(--color-primary);
	}
`;
