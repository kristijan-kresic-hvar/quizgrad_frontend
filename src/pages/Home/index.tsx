import {
	StyledHeader,
	StyledHeading,
	StyledHeroImg,
	StyledBlockQuote,
	StyledSectionLeft,
	StyledCtaButton,
} from './styled';
import { StyledFlexRow } from '../../shared/styled';
import heroPNG from '../../assets/img/hero.png';

const Home = () => {
	return (
		<StyledHeader>
			<StyledSectionLeft>
				<StyledHeading>
					Learn
					<br /> new concepts
					<br /> for each question
				</StyledHeading>
				<StyledBlockQuote>We help you prepare for exams and quizes</StyledBlockQuote>
				<StyledFlexRow mt={2.5}>
					<StyledCtaButton>Get Started</StyledCtaButton>
				</StyledFlexRow>
			</StyledSectionLeft>
			<StyledHeroImg src={heroPNG} alt="hero" />
		</StyledHeader>
	);
};

export default Home;
