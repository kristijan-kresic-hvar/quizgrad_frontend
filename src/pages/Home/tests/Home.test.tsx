//write test case for index file
import { render } from '@testing-library/react';
import Home from '../index';

describe('<Home />', () => {
	it('should render and match the snapshot', () => {
		const {
			container: { firstChild },
		} = render(<Home />);
		expect(firstChild).toMatchSnapshot();
	});
});
