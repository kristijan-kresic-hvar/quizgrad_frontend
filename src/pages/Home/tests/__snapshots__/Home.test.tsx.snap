// Vitest Snapshot v1

exports[`<Home /> > should render and match the snapshot 1`] = `
<header
  class="sc-bcXHqe cjleWV"
>
  <section
    class="sc-gswNZR lvHyg"
  >
    <h1
      class="sc-dkrFOg ccuukT"
    >
      Learn
      <br />
       new concepts
      <br />
       for each question
    </h1>
    <blockquote
      class="sc-eDvSVe iQJRIq"
    >
      We help you prepare for exams and quizes
    </blockquote>
    <div
      class="sc-gKPRtg CAv"
    >
      <button
        class="sc-jSUZER bKxzCm"
      >
        Get Started
      </button>
    </div>
  </section>
  <img
    alt="hero"
    class="sc-hLBbgP gmzKna"
    src="/src/assets/img/hero.png"
  />
</header>
`;
